homeassistant/components/mqtt/__init__.py

```diff
@@ -296,7 +296,11 @@ def async_setup(hass, config):
     if conf is None:
         conf = CONFIG_SCHEMA({DOMAIN: {}})[DOMAIN]
 
-    client_id = conf.get(CONF_CLIENT_ID)
+    # Remove the generation of the client_id after it's fixed in HBMQTT
+    import random
+    client_id = conf.get(
+        CONF_CLIENT_ID, 'home-assistant-{}'.format(random.randrange(0, 1000)))
```

Comment:
Should we check to make sure the protocol is 3.1.1?  I think with 3.1, zero length clientIds are supported by the pano-mqtt client.  That being said, it just generates a random clientId for you, but who know if someone depends on that.

Another option is to set the protocol to 3.1.